apps/frontend/src/pages/meeting-details/meeting-details.tsx

```diff
@@ -1,21 +1,196 @@
-import { Button, PlayerTrack } from "~/libs/components/components.js";
-import { useAppDispatch, useCallback, useParams } from "~/libs/hooks/hooks.js";
-import { actions as meetingActions } from "~/modules/meeting/meeting.js";
+import {
+	Button,
+	Icon,
+	Loader,
+	Markdown,
+	Navigate,
+	PlayerTrack,
+	SearchInput,
+} from "~/libs/components/components.js";
+import {
+	AppRoute,
+	DataStatus,
+	MeetingErrorMessage,
+	NotificationMessage,
+} from "~/libs/enums/enums.js";
+import { formatDate } from "~/libs/helpers/helpers.js";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useParams,
+	useSearchParams,
+} from "~/libs/hooks/hooks.js";
+import { notification } from "~/libs/modules/notifications/notifications.js";
+import { rehypeSanitize, remarkGfm } from "~/libs/plugins/plugins.js";
+import { DEFAULT_SEARCH_VALUE } from "~/modules/meeting-details/libs/default-values/meeting-details.default-values.js";
+import {
+	actions as meetingDetailsActions,
+	meetingDetailsApi,
+	sanitizeDefaultSchema,
+	searchInputValidationSchema,
+} from "~/modules/meeting-details/meeting-details.js";
+
+import styles from "./styles.module.css";
 
 const MeetingDetails: React.FC = () => {
 	const dispatch = useAppDispatch();
 	const { id } = useParams<{ id: string }>();
+	const [searchParameters] = useSearchParams();
+
+	const { dataStatus, meeting, transcription } = useAppSelector(
+		(state) => state.meetingDetails,
+	);
+	const { user } = useAppSelector((state) => state.auth);
+
+	const { control, errors } = useAppForm({
+		defaultValues: DEFAULT_SEARCH_VALUE,
+		validationSchema: searchInputValidationSchema,
+	});
+
+	useEffect(() => {
+		const sharedToken = searchParameters.get("token");
+		void dispatch(
+			meetingDetailsActions.getMeetingDetailsById({
+				id: Number(id),
+				sharedToken: sharedToken ?? undefined,
+			}),
+		);
+	}, [id, dispatch, searchParameters]);
+
+	const handleTranscriptionSearch = useCallback((value: string) => {
+		// TODO: implement handleTranscriptionSearch logic
+		return value;
+	}, []);
+
+	const handleShareClick = useCallback(() => {
+		if (!meeting || !meeting.id) {
```

Comment:
Should work like this
```suggestion
		if (!meeting?.id) {
```